### PR TITLE
Make use of eclipse/smarthome#4291

### PR DESCRIPTION
**Pull request by mueller-ma**
This commit enhances charts to support themes, scale legend according to
dpi and create option to hide legend

Signed-off-by: mueller-ma <mueller-ma@users.noreply.github.com>

See #373

### CODE REVIEW

**Comment by hreichert** [comment]
@mueller-ma That's pretty cool!

# Chart themes
:+1: Works good.


# Chart DPI
:+1: Works good.

Screenshot:
chart_app


# Chart legend
I'm not sure about the new setting for "Show chart legend".
The user can already set the `legend=true/false` attribute in his sitemap definition for the `Chart` element. But this gets ignored now. Maybe it's better to extend the `OpenHABWidget.java` and `OpenHAB2Widget.java` classes for this new sitemap attribute and take this setting to construct the chart URL. (refer to the 'period' attribute)

Here is a screenshot of the widgetJson from a sitemap with `Chart` with `legend=false`:
widgetjson

Thanks for your work!

**Comment by mueller-ma** [comment]
@hreichert I didn't know, that the legend attribute can be set in the sitemap.
Please test again. Also check what happens if no legend setting is set. The default should be true?

**Comment by mueller-ma** [comment]
> One note: As the default BasicUI uses the bright theme, I suggest to use the bright theme also for the HABDroid.Basic_ui theme.

Since this is my personal opinion that white looks better than bright, please take two screenshots and post it here and maybe in the forum and make a poll.

**Comment by hreichert** [comment]
@mueller-ma 
I will test later, but for now:
If the `legend` parameter is not set for a `Chart` item in the sitemap, the following is the default:
> Legend is automatically hidden, if there is only one chart series (-> one item) - otherwise the legend is shown

BasicUI theme: In my opinion, white looks also better and more modern. Right now I can not take screenshots, because my OH runs currently on the official snapshot distro, and the `white` chart theme is not yet integrated.

File: mobile/src/main/java/org/openhab/habdroid/model/OpenHABWidget.java [hunk 1]
```diff
@@ -25,7 +25,8 @@
 	private String type;
 	private String url;
 	private String period = "";
-    private String service = "";
+	private String service = "";
+	private boolean legend = true;
```
[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/model/OpenHABWidget.java]: I was aware of this problem, but couldn't find a good solution. Thanks!

**Comment by hreichert** [comment]
@mueller-ma Perfect, everything works as expected! Thanks again for your efforts.
FYI: Here is the PR for the `white` theme and some other optical improvements: https://github.com/eclipse/smarthome/pull/4368

**Comment by mueller-ma** [comment]
@hreichert Does the background of the charts always match the background of the app? Do you think a theme with a transparent background and bright font would be a good addition?

**Comment by digitaldan** [comment]
lgtm

**Pull request merged**